plugins/chat/spec/fabricators/chat_fabricator.rb

```diff
@@ -220,12 +220,16 @@
     thread.add(thread.original_message_user)
 
     if transients[:with_replies]
-      Fabricate.times(
-        transients[:with_replies],
-        :chat_message,
-        thread: thread,
-        use_service: transients[:use_service],
-      )
+      Fabricate
+        .times(
+          transients[:with_replies],
+          :chat_message,
+          thread: thread,
+          use_service: transients[:use_service],
+        )
+        .each { |message| thread.add(message.user) }
+
+      thread.update!(replies_count: transients[:with_replies])
```

Comment:
We were not adding the user answering to the thread as member of the thread which was a source of oddities in specs.

We were also not updating the replies_count which was also causing oddities.